### PR TITLE
Add ability to disable blacklisting

### PR DESCRIPTION
**Pull request by CalumMcCall**
- Move to injecting all backend config data into SE from UI
 - Tidy up handling and selection of backends
 - Allow users to specify custom blacklist url + disable blacklisting
 - Fix copying of string assets
 - Add copying of non-translatable strings(mostly URLs)
#### APK
[Download build #12287](http://192.168.120.33:8080/job/Pull%20Request%20Builder/12287/artifact/build/artifact/wire-dev-PR1968-12287.apk)
[Download build #12297](http://192.168.120.33:8080/job/Pull%20Request%20Builder/12297/artifact/build/artifact/wire-dev-PR1968-12297.apk)
[Download build #12299](http://192.168.120.33:8080/job/Pull%20Request%20Builder/12299/artifact/build/artifact/wire-dev-PR1968-12299.apk)
[Download build #12303](http://192.168.120.33:8080/job/Pull%20Request%20Builder/12303/artifact/build/artifact/wire-dev-PR1968-12303.apk)
[Download build #12304](http://192.168.120.33:8080/job/Pull%20Request%20Builder/12304/artifact/build/artifact/wire-dev-PR1968-12304.apk)
[Download build #12309](http://192.168.120.33:8080/job/Pull%20Request%20Builder/12309/artifact/build/artifact/wire-dev-PR1968-12309.apk)
[Download build #12311](http://192.168.120.33:8080/job/Pull%20Request%20Builder/12311/artifact/build/artifact/wire-dev-PR1968-12311.apk)
[Download build #12312](http://192.168.120.33:8080/job/Pull%20Request%20Builder/12312/artifact/build/artifact/wire-dev-PR1968-12312.apk)

### CODE REVIEW

File: app/src/main/scala/com/waz/zclient/appentry/AppEntryActivity.scala [hunk 1]
```diff
@@ -92,11 +92,13 @@ class AppEntryActivity extends BaseActivity {
     }.onUi(t => v.setText(t))
   }
 
-  ZMessaging.currentGlobal.blacklist.upToDate.onUi {
-    case false =>
-      startActivity(new Intent(this, classOf[ForceUpdateActivity]))
-      finish()
-    case _ =>
+  if (BuildConfig.ENABLE_BLACKLIST) {
+    ZMessaging.currentGlobal.blacklist.upToDate.onUi {
```
Comment (yeryomenkom) [inline app/src/main/scala/com/waz/zclient/appentry/AppEntryActivity.scala]: Maybe `Signal.ifFalse` in this case?

Comment (CalumMcCall) [reply app/src/main/scala/com/waz/zclient/appentry/AppEntryActivity.scala]: I'm not sure what you mean. I can't find any helper function called "ifFalse".

File: app/src/main/scala/com/waz/zclient/MainActivity.scala [hunk 2]
```diff
@@ -133,12 +133,14 @@ class MainActivity extends BaseActivity
...
     }
 
-    ZMessaging.currentGlobal.blacklist.upToDate.head.map {
-      case false =>
-        startActivity(new Intent(getApplicationContext, classOf[ForceUpdateActivity]))
-        finish()
-      case _ => //
-    } (Threading.Ui)
+    if (BuildConfig.ENABLE_BLACKLIST) {
+      ZMessaging.currentGlobal.blacklist.upToDate.head.map {
```
Comment (yeryomenkom) [inline app/src/main/scala/com/waz/zclient/MainActivity.scala]: Maybe `ifFalse` also here?

File: app/src/main/scala/com/waz/zclient/appentry/AppEntryActivity.scala (hunk 1 above)
Comment (marcoconti83) [reply app/src/main/scala/com/waz/zclient/appentry/AppEntryActivity.scala]: What about 'filter'? As suggested by @JPonte here: https://github.com/wireapp/wire-android/pull/1975/files#diff-1b098bed1af67295652680f8eef8059bR229

Comment (CalumMcCall) [reply app/src/main/scala/com/waz/zclient/appentry/AppEntryActivity.scala]: Nice idea! I've added that and I've merged both those areas into a static method which is a bit nicer.

**Comment by marcoconti83** [comment]
It would be better to split this PR in smaller PRs that address separate issues. I see that this PR addresses a few items:
- refactor: Firebase and backend configuration
- feature: add blacklisting feature flag
- feature: copy non-translatable string

File: app/src/main/scala/com/waz/zclient/MainActivity.scala [hunk 3]
```diff
@@ -455,5 +451,11 @@ object MainActivity {
   private def isSlideAnimation(oldTag: Option[String], newTag: String) = oldTag.fold(false) { old =>
     slideAnimations.contains((old, newTag)) || slideAnimations.contains((newTag, old))
   }
+
+  def checkBlacklist(activity: Activity)(implicit ecxt: EventContext): Unit =
```
Comment (JPonte) [inline app/src/main/scala/com/waz/zclient/MainActivity.scala]: Maybe find a better place to put this method. It doesn't really belong to the MainActivity itself, it makes it a bit confusing.

Comment (JPonte) [reply app/src/main/scala/com/waz/zclient/MainActivity.scala]: `ForceUpdateActivity` would make more sense for example.

**Pull request merged**